controllers/osbuild_controller.go

```diff
@@ -197,12 +198,25 @@ func (r *OSBuildReconciler) updateContainerComposeStatus(ctx context.Context, lo
 		return "", err
 	}
 
-	err = r.updateOSBuildConditionStatus(ctx, logger, osBuild, composeStatus, containerBuildDone, failedContainerBuild, startedContainerBuild)
+	status := composeStatus.Status
+	buildUrl := r.getBuildUrl(composeStatus)
+
+	err = r.updateOSBuildConditionStatus(ctx, logger, osBuild, status, containerBuildDone, failedContainerBuild, startedContainerBuild, buildUrl, emptyURL)
 	if err != nil {
 		logger.Error(err, "failed to update OSBuild condition status")
 		return "", err
 	}
-	return composeStatus, nil
+	return status, nil
+}
+
+func (r *OSBuildReconciler) getBuildUrl(composeStatus *composer.ComposeStatus) string {
+	buildUrl := emptyURL
+	imageStatuses := *(*composeStatus).ImageStatuses
```

Comment:
Do you need to dereference `composeStatus`? Wouldn't `*composeStatus.ImageStatuses` work? In addition you should check that it is not `nil` before dereferencing